### PR TITLE
Microsoft.VisualBasic.Core.Tests fail tests with NaN conversion

### PR DESCRIPTION
**Pull request by AlexRadch**
When the current culture is not English, then Microsoft.VisualBasic.Core.Tests fail tests with NaN conversion.

### CODE REVIEW

File: src/libraries/Microsoft.VisualBasic.Core/tests/SingleTypeTests.cs [hunk 1]
```diff
@@ -50,10 +57,15 @@ public void FromObject_ThrowsOverflowException(object value)
...
         {
-            Assert.Equal(expected, SingleType.FromString(value));
-            Assert.Equal(expected, SingleType.FromString(value, System.Globalization.NumberFormatInfo.InvariantInfo));
+            if (numberFormat is null)
+            {
+                Assert.Equal(expected, SingleType.FromString(value));
+                Assert.Equal(expected, SingleType.FromString(value, System.Globalization.NumberFormatInfo.InvariantInfo));
+            }
+            else
+                Assert.Equal(expected, SingleType.FromString(value, numberFormat));
```
Comment (danmoseley) [inline src/libraries/Microsoft.VisualBasic.Core/tests/SingleTypeTests.cs:68]:
```suggestion
            {
                Assert.Equal(expected, SingleType.FromString(value, numberFormat));
            }
```

**Review by danmoseley — APPROVED** [review]
thank you!

**Review by cston — APPROVED** [review]

**Comment by danmoseley** [comment]
@karelz there are several networking timeouts. are these just random endpoint issues ? eg 
```
❌ System.Net.Sockets.Tests.SendPacketsAsync.SendPacketsElement_FileStreamLargeOffset_Throws [Details] [History] [Artifacts]
Exception Message
Timed out
Expected: True
Actual:   False
CallStack
   at System.Net.Sockets.Tests.SendPacketsAsync.SendPackets(SendPacketsElement[] elements, SocketError expectedResult, Int32 bytesExpected, Byte[] contentExpected) in /_/src/libraries/System.Net.Sockets/tests/FunctionalTests/SendPacketsAsync.cs:line 693
   at System.Net.Sockets.Tests.SendPacketsAsync.SendPackets(SendPacketsElement element, SocketError expectedResult, Int32 bytesExpected) in /_/src/libraries/System.Net.Sockets/tests/FunctionalTests/SendPacketsAsync.cs:line 670
   at System.Net.Sockets.Tests.SendPacketsAsync.SendPacketsElement_FileStreamLargeOffset_Throws() in /_/src/libraries/System.Net.Sockets/tests/FunctionalTests/SendPacketsAsync.cs:line 535
```

**Comment by danmoseley** [comment]
@mattgal another very slow macOS build - a regular x64 build
https://dev.azure.com/dnceng/public/_build/results?buildId=1371594&view=logs&j=94422687-3fa3-516b-cf3b-23eb810befab&t=26102568-b21e-5f66-5226-c134f503e875

**Pull request merged**